FABRIC-AND-CFT.md

```diff
@@ -4,85 +4,85 @@ This page highlights the main differences (both technical and philosophical) bet
+CFT is a collection of Terraform modules and examples with opinionated GCP best practices implemented as individual modules for gradual adoption and off the shelf usage in organizations. CFT also includes an IaC [foundation repository](https://github.com/terraform-google-modules/terraform-example-foundation), though it's less developed compared to Fabric's.
 
 ## Key Differences
 
 <table>
   <tr>
    <td>
    </td>
-   <td><strong>Fabric</strong>
+   <td><strong>Fabric (FAST)</strong>
    </td>
-   <td><strong>CFT</strong>
+   <td><strong>CFT (Cloud Foundation Toolkit)</strong>
    </td>
   </tr>
   <tr>
    <td><strong>Target User</strong>
    </td>
-   <td>Organizations interested in forking, maintaining and customizing Terraform modules.
+   <td>Organizations interested in forking, maintaining, and customizing Terraform modules to have fine-grained control over their infrastructure.
    </td>
-   <td>Organizations interested in using opinionated, prebuilt Terraform modules.
+   <td>Organizations interested in using opinionated, prebuilt Terraform modules with a focus on Google Cloud best practices.
    </td>
   </tr>
   <tr>
    <td><strong>Configuration</strong>
    </td>
-   <td>Less opinionated allowing end users higher flexibility.
+   <td>Less opinionated, allowing end users higher flexibility and often offering more configuration options within a single module.
    </td>
-   <td>Opinionated by default, end users may need to fork if it does not meet their use case.
+   <td>Opinionated by default, end users may need to fork if it doesn't fully meet their use case.  Often requires composing multiple modules for complex configurations.
    </td>
   </tr>
   <tr>
    <td><strong>Extensibility</strong>
    </td>
-   <td>Built with extensibility in mind catering to fork and use patterns. Modules are often lightweight and easy to adopt / tailor to specific use cases.
+   <td>Built with extensibility in mind, catering to fork and use patterns. Modules are often lightweight and easy to adopt / tailor to specific use cases.
    </td>
-   <td>Not built with fork and use extensibility catering primarily to off the shelf consumption. Modules are tailored towards common usecases and extensible via composition.
+   <td>Not primarily built for fork-and-use extensibility, catering primarily to off-the-shelf consumption. Modules are tailored towards common use cases and extensible via composition.
    </td>
   </tr>
   <tr>
-   <td><strong>Config customization</strong>
+   <td><strong>Config Customization</strong>
    </td>
-   <td>Prefer customization using variables via objects, tight variable space.
+   <td>Prefer customization using variables via objects, providing a tight but comprehensive variable space within modules.
    </td>
-   <td>Prefer customization using variables via primitives.
+   <td>Prefer customization using variables via primitives, which can lead to more modularity but may require using multiple modules for customization.
```

Comment:
I don't see how more variables "can lead to more modularity but may require using multiple modules for customization"

Reply:
I was trying to convey that CFT seems to often prefer smaller, more focused modules, which *might* result in needing to use several of them together for complex configurations.  However, that's a design choice, not a direct consequence of variable count.

Should we remove that sentence altogether? Or perhaps rephrase it to focus solely on CFT's tendency towards smaller, more composable modules (if that makes sense), without mentioning variable count?

Reply:
> Should we remove that sentence altogether? Or perhaps rephrase it to focus solely on CFT's tendency towards smaller, more composable modules (if that makes sense), without mentioning variable count?

To be honest, I don't know. I don't understand what you're tying to highlight. 

You mention "CFT seems to often prefer smaller, more focused modules". IMO this is not true either. To give you an example, let's compare GKE modules

CFT (excluding documentation and boilerplate):
```
❯ tokei terraform-google-kubernetes-engine/*.tf terraform-google-kubernetes-engine/modules
===============================================================================
 Language            Files        Lines         Code     Comments       Blanks
===============================================================================
 HCL                   148        29025        21324         3226         4475
 Pan                     1           18           18            0            0
 Shell                  14          611          302          213           96
-------------------------------------------------------------------------------
 Markdown               18         3667            0         3151          516
 |- HCL                 10          716          622           11           83
 |- Shell                1            5            5            0            0
 |- YAML                 1            8            6            2            0
 (Total)                           4396          633         3164          599
===============================================================================
 Total                 181        33321        21644         6590         5087
===============================================================================
```

Fabric:
```
❯ tokei gke-cluster-autopilot gke-cluster-standard
===============================================================================
 Language            Files        Lines         Code     Comments       Blanks
===============================================================================
 HCL                     8         1893         1650          143          100
-------------------------------------------------------------------------------
 Markdown                2          276            0          200           76
 |- HCL                  2          359          335           24            0
 (Total)                            635          335          224           76
===============================================================================
 Total                  10         2169         1650          343          176
===============================================================================
```

Looking just at HCL, it's 29 kloc vs 1.8 kloc. I find hard it to justify that being smaller and more focused.

Reply:
Agreed with Julio. CFT is bloated, and prefers adding extra functionality via submodules which further complicates things.